### PR TITLE
Fix doc for denoise guassian

### PR DESCRIPTION
**Pull request by hmaarrfk**
## Checklist

- [ ] [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- [ ] Gallery example in `./doc/examples` (new features only)
- [ ] Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- [ ] Unit tests
- [ ] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/filters/rank/bilateral.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def sum_bilateral(image, selem, out=None, mask=None, shift_x=False,
 
     See also
     --------
-    skimage.filters.denoise_bilateral for a Gaussian bilateral filter.
+    skimage.filters.gaussian for a Gaussian bilateral filter.
```
Comment (jni) [inline skimage/filters/rank/bilateral.py]: This is incorrect. `filters.gaussian` is a standard Gaussian filter. See [wikipedia](https://en.wikipedia.org/wiki/Bilateral_filter) for the definition of a bilateral filter.

Probably this is meant to point to `skimage.restoration.denoise_bilateral`?

File: skimage/segmentation/tests/test_random_walker.py [hunk 2]
```diff
@@ -15,10 +15,8 @@
 PYAMG_MISSING_WARNING = r'pyamg|\A\Z'
 PYAMG_OR_SCIPY_WARNING = SCIPY_RANK_WARNING + '|' + PYAMG_MISSING_WARNING
 
-if (Version(np.__version__) >= '1.15.0'):
-    NUMPY_MATRIX_WARNING = 'matrix subclass'
-else:
-    NUMPY_MATRIX_WARNING = None
+# Certain version of numpy add this matrix warning, and others remove it.
+NUMPY_MATRIX_WARNING = r'matrix subclass|\A\Z'
```
Comment (jni) [inline skimage/segmentation/tests/test_random_walker.py]: Why are you changing this in this PR?

File: skimage/filters/rank/bilateral.py (hunk 1 above)
Comment (hmaarrfk) [reply skimage/filters/rank/bilateral.py]: good catch. I've been looking at my computer screen too long today.

File: skimage/segmentation/tests/test_random_walker.py (hunk 2 above)
Comment (hmaarrfk) [reply skimage/segmentation/tests/test_random_walker.py]: Trying to get green again. Will rollback.

**Comment by rfezzani** [comment]
Your fix is is correct, I tested it in #3850 and CI gone green again. THX

File: skimage/segmentation/tests/test_random_walker.py (hunk 2 above)
Comment (hmaarrfk) [reply skimage/segmentation/tests/test_random_walker.py]: The value of these maintenance PRs goes WAYYYYYYYYYYYY down if you don't want multiple fixes to come in the same PR.

Individually, these PRs are low priority, but the value they give to contributors is much higher if you consider them to be part of "getting automated tests to pass again". I ask that you reconsider your request to have seperated

Comment (jni) [reply skimage/segmentation/tests/test_random_walker.py]: Are you advocating for the VW approach to testing? ;)

I'm happy to have big bundled "let's fix everything" PRs. But every change needs to be the right change, not just masking all errors and warnings.

**Comment by jni** [comment]
Boom! =) Thanks @hmaarrfk

**Comment by hmaarrfk** [comment]
@meeseeksdev backport to v0.15.x

**Pull request merged**